subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/codecs/BindingsBackedCodec.kt

```diff
@@ -128,7 +128,9 @@ class BindingsBuilder {
         bind(type, SerializerCodec(serializer))
 
     fun bind(type: Class<*>, codec: Codec<*>) {
-        require(bindings.all { it.encodingForType(type) == null })
+        require(bindings.all { it.encodingForType(type) == null }) {
+            "There's already a binding that provide encoding for type '$type'"
```

Comment:
❌ _There's already an encoding for type '$type'_ ?

Reply:
Better thanks, fixed